python/idsse_common/idsse/common/publish_confirm.py

```diff
@@ -260,7 +271,7 @@ def on_delivery_confirmation(self, method_frame):
             '%i were acked and %i were nacked', self._message_number,
             len(self._deliveries), self._acked, self._nacked)
 
-    def publish_message(self, message, key=None):
+    def publish_message(self, message, key=''):
```

Comment:
If we're making tweaks to this method, it might be helpful for it to return a boolean of success/failure, or somehow tell the caller if a publish happened or not. 

Currently it is catching all exceptions and quietly logging them as warnings.